internal/storage/image.go

```diff
@@ -696,15 +696,15 @@ func (svc *imageService) UntagImage(systemContext *types.SystemContext, nameOrID
 			name = namedRef.DockerReference().String()
 		}
 
-		prunedNames := make([]string, 0, len(img.Names))
+		prunedNames := 0
 		for _, imgName := range img.Names {
 			if imgName != name && imgName != nameOrID {
-				prunedNames = append(prunedNames, imgName)
+				prunedNames += 1
 			}
 		}
```

Comment:
acutally, quick question: is there any risk to dropping this block and unconditionally running `svc.store.RemoveNames(...`?

Reply:
My first thought was to drop it but we need a way to check if any names left and if no delete image.